### PR TITLE
Fix #53 Add typescript definitions

### LINKED ISSUE #53: Typescript Definitions
Would be great if this plugin provided built-in `typings` like `ajv` does. Here is the declaration I'm using in my own projects:

```ts
// ajv-keywords.d.ts

declare module 'ajv-keywords' {
  import { Ajv } from 'ajv';
  type AdditionalKeywords =
    | 'typeof'
    | 'instanceof'
    | 'range'
    | 'exclusiveRange'
    | 'switch'
    | 'select'
    | 'selectCases'
    | 'selectDefault'
    | 'patternRequired'
    | 'prohibited'
    | 'deepProperties'
    | 'deepRequired'
    | 'uniqueItemProperties'
    | 'regexp'
    | 'formatMaximum'
    | 'formatMinimum'
    | 'formatExclusiveMaximum'
    | 'formatExclusiveMinimum'
    | 'dynamicDefaults';

  function keywords(ajv: Ajv, include?: AdditionalKeywords | AdditionalKeywords[]): void;
  export = keywords;
}
```

**BONUS:** it would be really great if this package also provided a module augmentation for `json-schema`. That way by importing this module, the additional keywords are made available on the interfaces exported by `json-schema`. Below is an (incomplete) example of what that would look like:

```ts
// json-schema.d.ts

import * as JsonSchema from 'json-schema';

declare module 'json-schema' {
  type BuiltinDynamics =
    | 'timestamp'
    | 'datetime'
    | 'date'
    | 'time'
    | 'random'
    | 'randomint'
    | 'seq';

  type DynamicDefault = BuiltinDynamics | string;
  type DynamicDefaultWithArgs = { func: DynamicDefault, [key: string]: any };

  interface KeywordAugmentations {
    dynamicDefaults?: {
      [key: string]: DynamicDefault | DynamicDefaultWithArgs;
    }

    // define other keywords...

  }

  export interface JSONSchema6 extends KeywordAugmentations { }
  export interface JSONSchema4 extends KeywordAugmentations { }
}
```

### PR DESCRIPTION
**Pull request by jsone-studios**
This pull request adds a typescript definitions file `ajv-keywords.d.ts` and therefore fixes #53.

### CODE REVIEW

**Comment by jsone-studios** [comment]
@epoberezkin friendly ping.

As the main [ajv](https://github.com/epoberezkin/ajv) package provides its own TypeScript declarations out of the box, it would be cool to do the same for ajv-keywords.

File: ajv-keywords.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    | 'patternRequired'
+    | 'prohibited'
+    | 'deepProperties'
+    | 'deepRequired'
+    | 'switch'
+    | 'select'
+    | 'dynamicDefaults'
+    ;
+
+  function keywords(ajv: Ajv, include?: AdditionalKeywords | AdditionalKeywords[]): Ajv;
```
Comment (epoberezkin) [inline ajv-keywords.d.ts]: I think it should be just `string | string[]`, Iet's please not make it the list of values.

Comment (jsone-studios) [reply ajv-keywords.d.ts]: I have just pushed the requested change

**Comment by epoberezkin** [comment]
merging it, it would be good to add TS tests in the same way as in ajv to avoid any syntax errors or typos

**Comment by epoberezkin** [comment]
Thanks for the reminder :) Will donut over the weekend.

**Comment by epoberezkin** [comment]
v3.5.0 is released now

**Pull request merged**